src/extractor/guidance/intersection_handler.cpp

```diff
@@ -283,6 +288,18 @@ void IntersectionHandler::assignFork(const EdgeID via_edge,
     }
 }
 
+void IntersectionHandler::assignTrivialTurns(const EdgeID via_eid,
+                                             Intersection &intersection,
+                                             const std::size_t begin,
+                                             const std::size_t end) const
+{
+    for (std::size_t index = begin; index != end; ++index)
+        if (intersection[index].entry_allowed)
+            intersection[index].turn.instruction = {
+                findBasicTurnType(via_eid, intersection[index]),
+                getTurnDirection(intersection[index].turn.angle)};
```

Comment:
We don't need the index, do we? What about

    for (auto& each : intersection)
      if (pred(each))
        each.attribute = {};